### PR TITLE
Remove unneded parameter permissions

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review. Its not the complete fix but an extraction of an improvement.

There are 3 call sides for `permitted_payment_attributes`

* Backend (only admins)
* Frontend (only customers)
* API (both, but in our case only admins)

This patch changes the `permitted_payment_attributes` to be most restrictive and enriches both non frontend call sides with the `amount` on demand.

I've got another patch in the pipeline that will remove the ability for users to access the API (But states, needed for non default country checkout), this will than cleanup the uncertainty.

File: api/app/controllers/spree/api/payments_controller.rb [hunk 1]
```diff
@@ -75,6 +75,10 @@ def perform_payment_action(action, *args)
         def payment_params
           params.require(:payment).permit(permitted_payment_attributes)
         end
+
+        def permitted_payment_attributes
+          super() + %i[amount]
```
Comment (dkubb) [inline api/app/controllers/spree/api/payments_controller.rb]: Does the api actually need to be able to modify the amount? Shouldn't this be something only an admin can do?

Comment (mbj) [inline api/app/controllers/spree/api/payments_controller.rb]: Yes. Removing regular user access for all API endpoints but `/api/states` (used for checkout) is in another branch.

This is about removing `amount` from the frontend.

Comment (dkubb) [inline api/app/controllers/spree/api/payments_controller.rb]: > Yes.

Oh, I did not know a non-admin user would need access to do this. It seems like a security risk.

Just FYI the api end-points are usable to browsers and non-admin users.

Comment (mbj) [inline api/app/controllers/spree/api/payments_controller.rb]: > Just FYI the api end-points are usable to browsers and non-admin users.

Yeah, I know. Hence another PR exist that removes API access to non-admins but whitelisted `/api/states` to remove the entire problem class.

**Pull request merged**